Yoast/wordpress-seo
# Fix breadcrumb page numbers in paginated series

## Context


* Page numbers in the breadcrumbs are useful, as they let users know where they are on the site.

## Summary


This PR can be summarized in the following changelog entry:

* Fixes a bug where the page number was not shown in the breadcrumb for paginated series.

## Relevant technical choices:

* Because date archives don't have an indexable permalink, only for this archive type we constructed the URL's from scratch. Since a "Page ..." breadcrumb was added, these URL's are now needed to turn the breadcrumb before the Page breadcrumb into a URL.
* The `is_paged` method doesn't detect pagination in the case of a paginated post. Therefore, we now also look at the `number_of_pages` in the indexables table. We do this for the last indexable in the `$indexables` array, because that is the indexable of the current page (the first indexable in that array, for example, would be the home page indexable).
* We created a helper function `get_current_page_number`. This function not only checks `\get_query_var( 'paged', 1 )` to find the pagination page number (this was already in the code before we started), but also checks `\get_query_var( 'page', 1 )`. Only the latter finds the page number on paginated posts.

## Test instructions

This PR can be tested by following these steps:

Read the instructions until the end before you start testing :) You need to test multiple things on each page.
* Make sure you have breadcrumbs enabled (see the issue).
* In paginated series (for example, on the home page, term archive pages or author archive pages), check that all pages except the first one have a Page breadcrumb. See the issue for an example. All the crumbs before the Page crumb should be URLs. The Page crumb should only be text.
* Check that the first page does NOT have a Page breadcrumb.
* On all the pages where you check the breadcrumbs, also check the Schema output. There should be a Schema graph piece called `BreadcrumbList`. The last breadcrumb ("Page ...") should NOT be output in the Schema.
    * Note that the homepage doesn't have a `BreadcrumbList` Schema piece.
* Pay extra attention to a date archive page. Check whether the URLs are working correctly for year, month and day archives.
* Pay extra attention to a paginated post. Create a new post that contains at least two Page break blocks to test this, and test both on pages 2 and 3 of that post.

## UI changes

## Documentation

## Quality assurance


Fixes https://yoast.atlassian.net/browse/P2-2
